module-4/backend/week_three.md

```diff
@@ -1,19 +1,54 @@
+Express is a fast, lightweight, unopinionated web application framework for Node.js. In the Ruby world, Express is similar to Sinatra, where it doesn't care about conventions very much so long as you string everything together. Also, Express is middleware.
+
 3. How do we setup a route when creating an API with Node and Express? Please provide a code snippet.
+Setting up a route when creating an API with Node and Express starts in the app.js file where you would define the route for a particular resource.
+
+```JavaScript
+var kittens = require('./routes/api/v1/kittens')
+```
+You would then continue to build out the directory structure for that specific resource within the routes directory.
+
 4. What do we use Knex for?
+Knex is a library that we use as a SQL builder to create queries in order to communicate with our database.
+
 5. How **could** you organize your code to follow the MVC design pattern in your Quantified Self project?
+You could organize your code to follow the MVC design pattern by adding model and controller directories to the parent directory. Within the model directory you would have files for each of the resources/objects that would be used throughout the applicatin and each would contain functions/methods applicable to the class/object. For the controller directory, there would be a contoller file for each of the routes and within each file would be action functions. Each of these action functions will define data that will be passed to and rendered in a view.
+
+For'xample
+```JavaScript
+router.get('/', function(req, res, next) {
+  kitten = {name: 'Sir FluffyButt'}
+  res.render('kittens', {kitten: kitten})
+})
+```
+
 6. How do you execute raw SQL in node?
+```Node
+database.raw('SELECT * FROM kittens')
+```
+
 7. What are some advantages to having your front end and back end code live in separate applications? What are some disadvantages?
+Advantages of having front end and back end code live in separate applications include being more agile in swapping out components of the infrastructure for newer and better technologies, having front and back end applications means deploy strategies are wholly separate, allowing you to minimize disruption to user experience during deploys and focus on the right tool chains, and lastly, another advantage of having a front and back end live in separate applications is the ability to change the application at different paces.
+
+Disadvantages of having front end and back end code live in separate applications involve the inability to take advantage of templating languages that frameworks may have, and having different strategies for build and deployment for both the front end and back end application - this may take up too much consumption of time and resources.
 
-#### Review  
+#### Review
 
 8. Describe DNS.
+DNS, or Domain Name Service, is the internet's system for converting aphabetic names into numeric IP addresses.
```

Comment:
I'd make sure you can go into a little more detail on this for interviews - we will go over in class!